include/libtorrent/torrent_handle.hpp

```diff
@@ -944,11 +946,11 @@ namespace libtorrent
 		// marks the piece with the given index as filtered
 		// it will not be downloaded
 		TORRENT_DEPRECATED
-		void filter_piece(int index, bool filter) const;
+		void filter_piece(piece_index_t index, bool filter) const;
```

Comment:
I was debating this. The reason I think it's fine is because when deprecated functions are enabled, ``piece_index_t`` is also implicitly convertible from and to ``int``, so it shouldn't make a significant difference.